src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp

```diff
@@ -82,26 +82,24 @@ RMSFusion::RMSFusion(uint64_t max_work_group_size) {
         }
 
         const auto& gamma_node = pattern_map.at(gamma).get_node_shared_ptr();
-        const auto& gamma_shape = gamma_node->get_output_partial_shape(0).to_shape();
 
         const auto& mean_node = pattern_map.at(mean).get_node_shared_ptr();
-        const auto & axes = pattern_map.at(mean_axes).get_node_shared_ptr();
+        const auto& axes = pattern_map.at(mean_axes).get_node_shared_ptr();
         auto axes_constant = std::dynamic_pointer_cast<ov::op::v0::Constant>(axes);
         auto axes_val = axes_constant->cast_vector<int64_t>();
         // allow last dimension only
-        if ((axes_val[0] != -1) && (axes_val[0] != (static_cast<int64_t>(mean_node->get_input_partial_shape(0).size()) - 1)))
+        if ((axes_val[0] != -1) &&
+            (axes_val[0] != (static_cast<int64_t>(mean_node->get_input_partial_shape(0).size()) - 1)))
             return false;
 
-        const int32_t vec_size = 8;
-        if (static_cast<int32_t>((gamma_shape.back() / vec_size)) > static_cast<int32_t>(max_work_group_size))
+        auto node = m.get_match_root();
+        if (transformation_callback(node)) {
```

Comment:
perhaps it's better to move this check to the beginning of this callback

Reply:
done